src/core/thread/address_resolver.cpp

```diff
@@ -529,24 +529,22 @@ void AddressResolver::HandleAddressQuery(Coap::Header &aHeader, Message &aMessag
 
     for (int i = 0; i < numChildren; i++)
     {
-        if (children[i].mState != Neighbor::kStateValid ||
-            (children[i].mMode & Mle::ModeTlv::kModeFFD) != 0 ||
-            children[i].mLinkFailures >= Mle::kFailedChildTransmissions)
+        if (children[i].GetState() != Neighbor::kStateValid ||
+            children[i].GetFullThreadDevice() ||
+            children[i].GetLinkFailures() >= Mle::kFailedChildTransmissions)
         {
             continue;
         }
 
-        for (int j = 0; j < Child::kMaxIp6AddressPerChild; j++)
+        for (uint8_t j = 0; j < Child::kMaxIp6AddressPerChild; j++)
         {
-            if (memcmp(&children[i].mIp6Address[j], targetTlv.GetTarget(), sizeof(children[i].mIp6Address[j])))
+            if (children[i].GetIp6Address(j) != *targetTlv.GetTarget())
```

Comment:
:+1: Another nice simplification of the compare automatically handling the 64-bit values.